File: Server/Domain/TimeSlot.cs
using Server.Domain.Auth;

namespace Server.Domain
{
    public class TimeSlot
    {
        public int Id { get; set; }
        public TimeOnly StartTime { get; set; }
        public TimeOnly EndTime { get; set; }
        public ICollection<Reservation> Reservations { get; set; }
        public int FacilityId { get; set; }
        public virtual Facility Facility { get; set; } = null!;
        public string CreatedById { get; set; } = null!;
        public User CreatedBy { get; set; }

        public TimeSlot()
        {
            Reservations = new List<Reservation>();
        }
    }
}

File: Client/src/models/timeSlot.ts
import { ReservationDto } from "./reservation";

type BaseTimeSlotDto = {
  startTime: Date;
  endTime: Date;
};

export type TimeSlotDto = {
  id: number;
  startTime: string;
  endTime: string;
  reservations: ReservationDto[];
  facilityId: number;
  createdById: string;
};

export type CreateTimeSlotDto = {
  startTime: string;
  endTime: string;
};

export type UpdateTimeSlotDto = {
  startTime: string;
  endTime: string;
};
